operator/v1/types_csi_cluster_driver.go

```diff
@@ -168,6 +168,64 @@ type AWSCSIDriverConfigSpec struct {
 	// +optional
 	KMSKeyARN string `json:"kmsKeyARN,omitempty"`
+
+	// efsVolumeMetrics sets the configuration for collecting metrics from EFS volumes used by the EFS CSI Driver.
+	// +openshift:enable:FeatureGate=AWSEFSDriverVolumeMetrics
+	// +optional
+	EFSVolumeMetrics *AWSEFSVolumeMetrics `json:"efsVolumeMetrics,omitempty"`
+}
+
+// AWSEFSVolumeMetricsState defines the modes for collecting volume metrics in the AWS EFS CSI Driver.
+// This can either enable recursive collection of volume metrics or disable metric collection entirely.
+// +kubebuilder:validation:Enum:="RecursiveWalk";"Disabled"
+type AWSEFSVolumeMetricsState string
+
+const (
+	// AWSEFSVolumeMetricsRecursiveWalk indicates that volume metrics collection in the AWS EFS CSI Driver
+	// is performed by recursively walking through the files in the volume.
+	AWSEFSVolumeMetricsRecursiveWalk AWSEFSVolumeMetricsState = "RecursiveWalk"
+
+	// AWSEFSVolumeMetricsDisabled indicates that volume metrics collection in the AWS EFS CSI Driver is disabled.
+	AWSEFSVolumeMetricsDisabled AWSEFSVolumeMetricsState = "Disabled"
+)
+
+// AWSEFSVolumeMetrics defines the configuration for volume metrics in the EFS CSI Driver.
+// +union
+type AWSEFSVolumeMetrics struct {
+	// state defines the state of metric collection in the AWS EFS CSI Driver.
+	// This field is required and must be set to one of the following values:
+	// - Disabled: No metrics collection will be performed. This is the default value.
+	// - RecursiveWalk: The AWS EFS CSI Driver will recursively scan volumes to collect metrics.
+	//   This process may result in high CPU and memory usage, depending on the volume size.
+	// +unionDiscriminator
+	// +kubebuilder:default:=Disabled
+	// +default="Disabled"
+	// +optional
+	State AWSEFSVolumeMetricsState `json:"state,omitempty"`
+
+	// recursiveWalk provides additional configuration for collecting volume metrics in the AWS EFS CSI Driver
+	// when the state is set to RecursiveWalk.
+	// +unionMember,optional
+	RecursiveWalk *AWSEFSVolumeMetricsRecursiveWalkConfig `json:"recursiveWalk,omitempty"`
+}
+
+// AWSEFSVolumeMetricsRecursiveWalkConfig defines options for volume metrics in the EFS CSI Driver.
+type AWSEFSVolumeMetricsRecursiveWalkConfig struct {
+	// refreshPeriodMinutes specifies the frequency, in minutes, at which volume metrics are refreshed.
+	// If not specified, the default refresh period in the EFS CSI Driver is 240 minutes.
+	// The valid range is from 1 to 43200 minutes (30 days).
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:validation:Maximum=43200
+	RefreshPeriodMinutes int32 `json:"refreshPeriodMinutes,omitempty"`
+
+	// fsRateLimit defines the rate limit, in goroutines per file system, for processing volume metrics.
+	// If not specified, the default rate limit in the EFS CSI Driver is 5 goroutines per file system.
+	// The valid range is from 1 to 100 goroutines.
+	// +kubebuilder:validation:Required
```

Comment:
As above, I think make this optional and add the wording I suggested to show what happens when it is omitted

Reply:
Done.